tests/caching/metadata_store/test_base.py

```diff
@@ -0,0 +1,194 @@
+from typing import Dict
+
+import pytest
+
+from hamilton.caching.cache_key import create_cache_key
+from hamilton.caching.fingerprinting import hash_value
+from hamilton.caching.stores.memory import InMemoryMetadataStore
+from hamilton.caching.stores.sqlite import SQLiteMetadataStore
+from hamilton.graph_types import hash_source_code
+
+
+def _instantiate_metadata_store(metadata_store_cls, tmp_path):
+    if metadata_store_cls == SQLiteMetadataStore:
+        return SQLiteMetadataStore(path=tmp_path)
+    elif metadata_store_cls == InMemoryMetadataStore:
+        return InMemoryMetadataStore()
+    else:
+        raise ValueError(
+            f"Class `{metadata_store_cls}` isn't defined in `_instantiate_metadata_store()`"
+        )
+
+
+def _mock_cache_key(
+    node_name: str = "foo",
+    code_version: str = "FOO-1",
+    dependencies_data_versions: Dict[str, str] = None,
+) -> str:
+    """Utility to create a valid cache key from mock values.
+    This is helpful because ``code_version`` and ``data_version`` found in ``dependencies_data_versions``
+    must respect specific encoding.
+    """
+    dependencies_data_versions = (
+        dependencies_data_versions if dependencies_data_versions is not None else {}
+    )
+    return create_cache_key(
+        node_name=node_name,
+        code_version=hash_source_code(code_version),
+        dependencies_data_versions={k: hash_value(v) for k, v in dependencies_data_versions},
+    )
+
+
+@pytest.fixture
+def metadata_store(request, tmp_path):
+    metdata_store_cls = request.param
+    metadata_store = _instantiate_metadata_store(metdata_store_cls, tmp_path)
+
+    yield metadata_store
+
+    metadata_store.delete_all()
+
+
+@pytest.mark.parametrize(
+    "metadata_store",
+    [SQLiteMetadataStore, InMemoryMetadataStore],
```

Comment:
Nice parameterizatio -- we should have a quick guide on "implementing your own" (noting it's a bit experimental...)

Reply:
Not a blocker

Reply:
The idea of `test_base.py` is that all implementations should be passing. Currently, all test have a parameterization with the list of implementations.

I think it would makes sense to have a constant at the top of the module `IMPLEMENTATIONS = [...]` containing the list of classes and using this list throughout.